lib/spack/docs/index.rst

```diff
@@ -49,6 +49,12 @@ or refer to the full manual below.
 
 .. toctree::
    :maxdepth: 2
+   :caption: Module File Support
+
+   module_file_support
```

Comment:
I agree that Module File Support deserves it's own file, but I'm not sure if we should have an entire top-level toctree just for one file.